src/NuGet.Clients/NuGet.VisualStudio.Client/.vsixignore

```diff
@@ -4,6 +4,9 @@ Ben.Demystifier.dll
 MessagePack.Annotations.dll
 MessagePack.dll
 Microsoft.Bcl.AsyncInterfaces.dll
+Microsoft.Extensions.FileProviders.Abstractions.dll
+Microsoft.Extensions.FileSystemGlobbing.dll
+Microsoft.Extensions.Primitives.dll
```

Comment:
Does msbuild restore on the command line still work?   VS's PrivateAssemblies aren't used by MSBuild.exe.

Reply:
That's a good question, do any of our end-to-end tests cover that scenario?

Reply:
No. You'll need to create a VM, either install the CI build's bootstrapped VS, or just "normal VS" and download the NuGet vsix and install it, then manually test it.

It probably would be a good idea to add a test that runs msbuild.exe at some point. Thinking about how to make such a test work, especially locally where I have many versions of VS installed side-by-side, gives me a headache.

Reply:
I thought we had MSBuild end to end tests, https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Core.FuncTests/Msbuild.Integration.Test/MsbuildRestoreTaskTests.cs.

Reply:
Didn't the assemblies get moved to the `PublicAssemblies` recently?

Reply:
It looks like the functional tests use the locally built task assembly which is next to the new dependencies so it "works".  I'm testing with the VSIX and will report back.

Reply:
🤦‍♂️ 
```
D:\temp\ClassLibrary2>msbuild /t:restore /clp:v=m;summary;forcenoalign
MSBuild version 17.5.0-preview-22564-01+d14b74d6f for .NET Framework
  Determining projects to restore...
D:\VS-IntPreview\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.targets(132,5): error : Could not load file or assembly 'Microsoft.Extensions.FileProviders.Abstractions, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' or one of its dependencies. The system cannot find the file specified. [D:\temp\ClassLibrary2\ClassLibrary2.csproj]

Build FAILED.

D:\VS-IntPreview\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.targets(132,5): error : Could not load file or assembly 'Microsoft.Extensions.FileProviders.Abstractions, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' or one of its dependencies. The system cannot find the file specified. [D:\temp\ClassLibrary2\ClassLibrary2.csproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:01.62
```

Reply:
Ask me how I knew that MSBuild doesn't load assemblies from PublicAssemblies or PrivateAssemblies. Go on, I dare you 😁 

Yeah, unfortunately we have to make our MSBuild tasks entirely "self contained". (or in our msbuild task code, I guess we could subscribe to the app domain assembly load event, and redirect specific assemblies individually? Though, that might break restore on mono 🤷‍♂️).

I think if we make sure the versions of the dependencies we use are the same, or lower, than in public/private assemblies, then VS's binding redirects should load it from there, so we shouldn't get multiple copies of the same assembly in memory, unless we use a higher version of the assemblies.

But I never looked into why the previous attempt that this fix failed RPS, so I don't know what to look for in this PR to avoid a repeat of the previous attempt.

Reply:
It was a great catch.  I'm starting to think our build task should be ILMerge'd and in a different directory while all of the actual assemblies our VSIX needs should be in the same old spot.  It feels like we're swimming upstream with our task assembly in here.

Reply:
ilmerge is abandonware. we should not use it for anything. One reason is that it doesn't support newer features like "portable" pdbs or "public signing", which causes additional technical debt on us. I will actively resist any attempt to increase use of ilmerge. The only time I want to read the name ilmerge is in PRs that remove it from our repo. Unfortunately, I see no alternative for nuget.exe, but in my opinion nothing else should use it.

Yes, maybe we need to split our VS extension and msbuild tasks into different (sub)directories (but no need for ilmerge), although I'm not sure that'll actually help us in this situation, unless we also duplicate all the assemblies that are shared, but then our VSIX size will bloat.

I can't see an ideal/perfect solution 😞

Reply:
Do you know why our Pack task is ILMerged?  I'd love to know more history about that one.

One way to get NuGet.exe to not be ILMerged is to actually compile the sources directly into it which would be a fair bit of work but totally possible.  Another way would be to have it be a self-extracting ZIP that then runs the expanded EXE with the DLLs next to it.  

The only other way I can think of to avoid shipping these is to update `MSBuild.exe.config` to find these assemblies in the shared location.

Reply:
> Do you know why our Pack task is ILMerged? I'd love to know more history about that one.

At some point it was causing assembly loading issues as the pack tasks don't have to match what the restore tasks carry, as pack can be installed as a separate SDK/package. 

That was so long ago, that I haven't found much details on GH, so at this point I don't remember the exact scenario so I can't really retest. 

Thinking through the scenario, I think we might be safe to stop ilmerging, but that's just a potentially risky change that hasn't been prioritized.